src/codegen.cpp

```diff
@@ -3829,8 +3841,8 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
             }
         }
         else if (fld.typ == (jl_value_t*)jl_symbol_type) {
-            if (jl_is_datatype(utt) && !jl_is_namedtuple_type(utt)) { // TODO: Look into this for NamedTuple
-                if (jl_struct_try_layout(utt) && (jl_datatype_nfields(utt) == 1)) {
+            if (jl_is_datatype(utt) && (utt != jl_module_type) && jl_struct_try_layout(utt)) {
+                if ((jl_datatype_nfields(utt) == 1 && !jl_is_namedtuple_type(utt))) {
```

Comment:
I'm assuming that since you're removing the TODO, you have come to understand why the single-field struct case doesn't apply to NamedTuples?

Can you please leave a comment here to replace the Todo and explain WHY the new code is excluding named tuples?

I'd also appreciate a comment on why we special case the 1-field struct at all.

Reply:
So what this change does is extend the optimization I did on the last PR to non single-field structs/named-tuples. The codegen isn't as good because it's not known which field we are accessing, though I wonder if it's cleaner to remove the special case because `emit_getfield_unknownidx` already has a special case checker. What do you think @vtjnash